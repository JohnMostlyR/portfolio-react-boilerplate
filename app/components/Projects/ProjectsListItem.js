import styled from 'styled-components';
import { lighten, darken } from 'polished';
import { theme } from '../../styles/theme';
import { BASE_LINE_HEIGHT } from '../../styles/typography';

const { backgroundColor, color } = theme.project;
const ARROW_ODD_PATH_ONE_COLOR = lighten(0.1, backgroundColor.odd).substring(1);
const ARROW_ODD_PATH_TWO_COLOR = darken(0.1, backgroundColor.odd).substring(1);
const ARROW_ODD_PATH_THREE_COLOR = lighten(0.1, backgroundColor.odd).substring(
  1,
);
const ARROW_EVEN_PATH_ONE_COLOR = lighten(0.1, backgroundColor.even).substring(
  1,
);
const ARROW_EVEN_PATH_TWO_COLOR = darken(0.1, backgroundColor.even).substring(
  1,
);
const ARROW_EVEN_PATH_THREE_COLOR = lighten(
  0.1,
  backgroundColor.even,
).substring(1);
const ARROW_ODD = `data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 427.9 175.1 132.1' preserveAspectRatio='xMinYMid'%3E%3Cpath stroke='%23${ARROW_ODD_PATH_ONE_COLOR}' fill='%23${ARROW_ODD_PATH_ONE_COLOR}' d='M0 427.9l175.1 65.7-43.7-65.7z'/%3E%3Cpath fill='%23${ARROW_ODD_PATH_TWO_COLOR}' d='M175.1 493.6l-87.5 21.9v-54.8'/%3E%3Cpath fill='%23${ARROW_ODD_PATH_THREE_COLOR}' d='M87.6 515.5l21.2 44.5v-49.8'/%3E%3C/svg%3E`;
const ARROW_EVEN = `data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 427.9 175.1 132.1' preserveAspectRatio='xMinYMid'%3E%3Cpath stroke='%23${ARROW_EVEN_PATH_ONE_COLOR}' fill='%23${ARROW_EVEN_PATH_ONE_COLOR}' d='M0 427.9l175.1 65.7-43.7-65.7z'/%3E%3Cpath fill='%23${ARROW_EVEN_PATH_TWO_COLOR}' d='M175.1 493.6l-87.5 21.9v-54.8'/%3E%3Cpath fill='%23${ARROW_EVEN_PATH_THREE_COLOR}' d='M87.6 515.5l21.2 44.5v-49.8'/%3E%3C/svg%3E`;

const ProjectsListItem = styled.li`
  margin: 0;
  padding: 0;
  position: relative;
  border-radius: 0.5rem;
  box-shadow: 0 0 10px grey;
  transition: box-shadow 0.2s ease-in-out;

  @media (min-width: 900px) {
    box-shadow: none;
    filter: drop-shadow(0 0 5px grey);
    transition: filter 0.2s ease-in-out;
  }

  &:not(:first-child) {
    margin-top: ${BASE_LINE_HEIGHT * 2}rem;
  }

  &:nth-child(odd) {
    background-color: ${backgroundColor.odd};
    color: ${color.odd};

    @media (min-width: 900px) {
      margin-left: 14vw;
      border-bottom-right-radius: 0;

      &::after {
        content: '';
        height: 9vmin;
        left: 0;
        margin-top: -1px;
        position: absolute;
        right: 0;
        top: 100%;
        z-index: 1;
        background-image: ${() => `url("${ARROW_ODD}")`};
        background-repeat: no-repeat;
        background-size: contain;
        background-position: left top;
        transform: rotateY(180deg);
      }
    }
  }

  &:nth-child(even) {
    background-color: ${backgroundColor.even};
    color: ${color.even};

    @media (min-width: 900px) {
      margin-right: 14vw;
      border-bottom-left-radius: 0;

      &::after {
        content: '';
        height: 9vmin;
        left: 0;
        margin-top: -1px;
        position: absolute;
        right: 0;
        top: 100%;
        z-index: 1;
        background-image: ${() => `url("${ARROW_EVEN}")`};
        background-repeat: no-repeat;
        background-size: contain;
        background-position: left top;
      }
    }
  }
`;

export default ProjectsListItem;
