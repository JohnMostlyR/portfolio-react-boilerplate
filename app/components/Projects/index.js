import React from 'react';
import PropTypes from 'prop-types';

import ProjectsList from './ProjectsList';
import ProjectsListItem from './ProjectsListItem';

import Project from '../../components/Project';

function Projects({ projects, search }) {
  return (
    <React.Fragment>
      <ProjectsList>
        {projects.map(({ title, description, thumbnail }, idx) => (
          <ProjectsListItem key={title}>
            <Project
              detailsBodyText={description}
              isOdd={!(idx % 2)}
              search={search}
              thumbnail={thumbnail}
              title={title}
            />
          </ProjectsListItem>
        ))}
      </ProjectsList>
    </React.Fragment>
  );
}

Projects.propTypes = {
  projects: PropTypes.array,
  search: PropTypes.string,
};

Projects.defaultProps = {
  projects: [],
};

export default Projects;
