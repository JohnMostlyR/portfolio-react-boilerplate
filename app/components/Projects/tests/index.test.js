import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import Projects from '../index';

import { parsedContentfulData } from '../../../containers/ProjectsPage/tests/parsedContentfulData';

describe('<Projects />', () => {
  it('should render and match the snapshot', () => {
    const renderedComponent = shallow(
      <Projects projects={parsedContentfulData} />,
    );
    expect(toJson(renderedComponent)).toMatchSnapshot();
  });
});
