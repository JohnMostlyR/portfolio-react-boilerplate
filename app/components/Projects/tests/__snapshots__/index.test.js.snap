// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Projects /> should render and match the snapshot 1`] = `
<React.Fragment>
  <ProjectsList>
    <ProjectsListItem
      key="This is the title field for Project 1"
    >
      <Project
        detailsBodyText="This is the description field Project 1"
        isOdd={true}
        thumbnail={
          Object {
            "file": Object {
              "contentType": "image/png",
              "details": Object {
                "image": Object {
                  "height": 230,
                  "width": 345,
                },
                "size": 22495,
              },
              "fileName": "medium-m.png",
              "url": "/medium-m.png",
            },
            "title": "Presentation showing how this project looks on mobile, tablet and desktop screens",
          }
        }
        title="This is the title field for Project 1"
      />
    </ProjectsListItem>
    <ProjectsListItem
      key="This is the title field for Project 2"
    >
      <Project
        detailsBodyText="This is the description field Project 2"
        isOdd={false}
        thumbnail={
          Object {
            "file": Object {
              "contentType": "image/png",
              "details": Object {
                "image": Object {
                  "height": 230,
                  "width": 345,
                },
                "size": 22495,
              },
              "fileName": "medium-m.png",
              "url": "/medium-m.png",
            },
            "title": "Presentation showing how this project looks on mobile, tablet and desktop screens",
          }
        }
        title="This is the title field for Project 2"
      />
    </ProjectsListItem>
  </ProjectsList>
</React.Fragment>
`;
